### ISSUE TITLE
When a previously picked line is voided in loft, the RF_PICK table is not being updated.

### ISSUE THREAD

**Issue by kentbower** [kbower/tickettest1#2311]
_Imported from trac ticket 2311.
Created by **dwarren**
Opened in trac: May 1, 2013, 5:20:31 PM
Last modified in trac: May 15, 2013, 10:12:00 AM_
***
_by dwarren:_
Please see the attached copies for a better understanding.
Recently discovered issues where a RF pick could not be generated for a line that was previously voided now reserved on a new sale.
This might require a call to get a better understanding of the details if needed.
Basically, when a previously picked line is voided in loft, the RF_PICK table is not being updated.

 

***
* sjohnson attached [rf_pick.pdf](https://trac.retailarchitects.com/trac/attachment/ticket/2311/rf_pick.pdf) on May 1, 2013, 5:21:04 PM

**Comment by kentbower** [comment]
_**kbower** trac comment on May 1, 2013, 6:19:14 PM:_

I suspect a similar problem exists when such a line is unfilled. 

Kent

**Comment by kentbower** [comment]
_**kbower** trac comment on May 9, 2013, 2:49:08 PM:_

Mark, Loft currently isn't even aware of the RF_PICK table, so it certainly isn't doing whatever it needs to in that table.

The pdf mentioned something about STAGE and INV_XREF. _[the pdf only mentions that after setting the void flag and re-running PKR, the PKR can properly update the INV_XREF's STAGE](Edit:)_

Please enumerate out exactly what changes related to RF need to take place when a line is a) unfilled and b) voided.  None of these are programmed currently.
___
This was a surprise for us also.

I would say that if a line of an order is unfilled or voided and it exists in the RF_PICK table, then the void flag needs to be changed to “Y”.

This holds true for if an order is removed from a truck.  We un-truck in V12, so that part is working properly.
___
Does the void flag in RF_PICK mean the line is voided or the pick is voided?

Have you double checked when E1 simply unfills a line with a pick that this occurs?
___
Mark email
_Kent_
**Mark's reply**:
Getting back to this.   I believe I went through every scenario possible to resolve the issues with updating the RF_PICK table:

Everything is pertaining to when a pick should be voided.

Currently in E1, if an order is untrucked and a delivery date is changed, the void flag in RF_PICK is set to “Y”

_Since you said " if an order is untrucked and a delivery date is changed", am I to understand that untrucking alone or changing the delivery date alone does not set the void flag to "Y"?_ **Sorry for the confusion**
_Does untrucking alone set the void flag?_ **YES**
_Does changing the delivery date alone set the void flag?_ **YES**


If a line that was on a truck (exists in the RF_PICK table) is split in Loft, the Void flag needs changed to “Y”.

_When you say the "line is split," I think you are talking about when the order is split and the line moved to the new split order, is that correct?_ **Yes.  that is correct.**
_Also, is it possible for an rf_pick to exist for lines that are not trucked, or does E1 absolutely require the order to be trucked for the PKR to generate a pick?  (Even for customer pick ups?)_ **There is currently something weird with Pickups that we don’t understand and it will generate a pick for a pickup at times but it isn’t when we want to the pick to generate.  I believe it happens if something is scanned in the warehouse and that RFID is connected to a Pickup that is set for that days pickup date.  We only want to generate picks for trucked orders, including pickups.**

If a SAL is Voided and the lines on the voided SAL exist in the RF_PICK table, the void flag needs changed to “Y”

If a line is unreserved and that line exists in the RF_PICK table, the void flag needs changed to “Y”

When an order gets finalized the RF_PICK record is updated to “Y” for the void flag.


Everything is driven by RFID in the RF_PICK table.  If the RFID in the RF_PICK record does not show  a “Y” for the void flag and the RFID gets reserved to a different sales order or on the same order, the Pick will not create because the RFID already exists to be picked for a previous pick date.
 
I know you will have additional questions, but I am currently controlling this manually when a pick in the DC cannot be generated.

**Comment by kentbower** [comment]
_**kbower** trac comment on May 10, 2013, 9:11:56 AM:_

Mark, another question:
When E1 updates the void flag and sets it to 'Y' does it also update the EMP_CD?
___
_No that remains the same.  That EMP_CD is the person that generated the pick through the PKR._

_On a side note, when you get to IST’s, the same things happen with RF_PICK_

**Comment by kentbower** [comment]
_**kbower** trac comment on May 10, 2013, 9:51:04 AM:_

So I'm wondering whether Loft should check for and void the pick anytime a line is moved to a split order or if it only needs to make that check if the order is trucked.  I understand Morris only generates picks for trucked orders, but I'm concerned about getting the code right for all future Loft retailers.  So, if GERS allows picks to be generated even for untrucked orders then Loft should be conservative and make the check even for untrucked orders.

Does E1 allow RF picks to be generated for untrucked orders?  
___
_I believe it should always check if the record exists since I did verify that you are able to generate a pick without being on a truck._

_It would be my suggestion that there be a security/parameter  added to allow/disallow a pick to be generated if trucked or not trucked._

**Comment by kentbower** [comment]
_**kbower** trac comment on May 10, 2013, 10:08:58 AM:_

I think this boils down to these rules:
If `rf_enabled` parameter then set picks to voided = 'Y' for these cases, for SAL*:
 * all on order:
   * order's delivery date changes
   * order's status changes ('F' OR 'V'), but only 'F' is important, since the unreservation of others with void will catch those. See [#2331](2331)
   * order is untrucked (stop deleted)
 * specific to rf_id: 
   * line changes orderid, even partly
   * line unreserved, even partly

If line is being broken, picks with RF_IDs should move to new line.

_* From Morris: "We do not use the EEX function because we were told when we went RF, that we couldn’t use EEX any longer. SER orders are done manually and doesn’t really pertain to inventory."_

**Comment by kentbower** [comment]
_**kbower** trac comment on May 14, 2013, 11:09:34 AM:_

(Changeset [[4134]](../commit/60277c16ea8230d3703de51c155c4b031233929f))
* closes #2311, When a previously picked line is voided in loft, the RF_PICK table is not being updated. #mor

**Comment by kentbower** [comment]
_**kbower** trac status change on May 14, 2013, 11:09:34 AM:_ **closed**

**Comment by kentbower** [comment]
_**kbower** trac comment on May 14, 2013, 4:20:45 PM:_

(Changeset [[4138]](../commit/e8db7720491d0bbab87e024cb38b94f237aa6de1))
* re #2311, try to get out of the way of GERS RF_PICKS triggers #mor

**Comment by kentbower** [comment]
_**kbower** trac comment on May 14, 2013, 5:23:22 PM:_

Rewrite of rfids updates needed now because of gers triggers.

**Comment by kentbower** [comment]
_**kbower** trac status change on May 14, 2013, 5:23:22 PM:_ **reopened**

**Comment by kentbower** [comment]
_**kbower** trac comment on May 15, 2013, 10:12:00 AM:_

(Changeset [[4141]](../commit/5ae54d242f43f9d6ee3cbcb87c96cdb48d5ff4db))
* fixes #2311 regarding GERS RF_PICK table triggers interaction

**Comment by kentbower** [comment]
_**kbower** trac status change on May 15, 2013, 10:12:00 AM:_ **closed**
